bin/ch/Helpers.cpp

```diff
@@ -5,10 +5,63 @@
 #include "stdafx.h"
 #include "Codex/Utf8Codex.h"
 
-HRESULT Helpers::LoadScriptFromFile(LPCWSTR filename, LPCWSTR& contents, bool* isUtf8Out, LPCWSTR* contentsRawOut, UINT* lengthBytesOut, bool printFileOpenError)
+///
+/// Use the codex library to encode a UTF16 string to UTF8.
+/// The caller is responsible for freeing the memory, which is allocated
+/// using malloc.
+/// The returned string is null terminated.
+///
+HRESULT Helpers::WideStringToNarrowDynamic(LPCWSTR sourceString, LPSTR* destStringPtr)
+{
+    size_t cchSourceString = wcslen(sourceString);
+    size_t cbDestString = (cchSourceString + 1) * 3;
+    
+    utf8char_t* destString = (utf8char_t*)malloc(cbDestString);
+    if (destString == nullptr)
+    {
+        return E_OUTOFMEMORY;
+    }
+    if (cchSourceString >= MAXUINT32)
+    {
+        return E_OUTOFMEMORY;
+    }
```

Comment:
can we move this check above - before we do malloc. Also if we can check the cbDestString as well.